cmd/geth/main.go

```diff
@@ -347,6 +351,15 @@ func geth(ctx *cli.Context) error {
 	stack, backend := makeFullNode(ctx)
 	defer stack.Close()
 
+	if ctx.GlobalBool(utils.RunHeimdallFlag.Name) {
+		shutdownCtx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+		defer stop()
```

Comment:
don't we cancel the context and HeimdallService by the end of `geth()` call?

Reply:
stop() is same as calling ctx.cancel()
That's why I used `defer stop()`
My understanding is this will call stop at the end of parent function which is `geth()` in this case.

Reply:
It will stop by the end of the current function. Let's recheck it with a small example in play.go. could you make one?

Reply:
Yeah but it will only reach end of this function when the node is supposed to stop as `stack.Wait()` will keep it waiting.

Reply:
have you tested that it works fine? like with term signal or something?

Reply:
Yup, I have added few logs as well in graceful shutdown of all the services and it worked fine while testing.